mhs/mhs/common/state/persistence_adapter.py

```diff
@@ -1,8 +1,37 @@
 """
 This module defines the state adapter interface, used to allow support for multiple state database implementations.
 """
+from typing import Optional
```

Comment:
I said above we're avoiding importing specific classes in favor of whole modules but just as a note this doesn't apply to the typing class so this is fine